x/launch/types/message_request_add_account_test.go

```diff
@@ -0,0 +1,65 @@
+package types_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/spn/testutil/sample"
+	"github.com/tendermint/spn/x/launch/types"
+)
+
+func TestMsgRequestAddAccount_ValidateBasic(t *testing.T) {
+	addr := sample.AccAddress()
+	chainID, _ := sample.ChainID(10)
+	tests := []struct {
+		name string
+		msg  types.MsgRequestAddAccount
+		err  error
+	}{
+		{
+			name: "invalid address",
+			msg: types.MsgRequestAddAccount{
+				Address: "invalid_address",
+				ChainID: chainID,
+				Coins:   sample.Coins(),
+			},
+			err: sdkerrors.Wrap(sdkerrors.ErrInvalidAddress,
+				"invalid address (invalid_address): decoding bech32 failed: invalid index of 1"),
+		}, {
+			name: "invalid chain id",
+			msg: types.MsgRequestAddAccount{
+				Address: sample.AccAddress(),
+				ChainID: "invalid_chain",
+				Coins:   sample.Coins(),
+			},
+			err: sdkerrors.Wrap(types.ErrInvalidChainID, "invalid_chain"),
+		}, {
+			name: "message without coins",
+			msg: types.MsgRequestAddAccount{
+				Address: addr,
+				ChainID: chainID,
+				Coins:   sdk.NewCoins(),
+			},
+			err: sdkerrors.Wrap(types.ErrInvalidCoins, addr),
+		}, {
+			name: "valid message",
```

Comment:
We can have a test as well for invalid coin by using `Coins: sdk.NewCoins(sdk.Coin{
  Amount: sdk.NewInt(10),
  Denom: "",
}),`

Reply:
Invalid coin venom causes panic to the cosmos SDK library. we must change the test to recover and assert the panic message only for this case:

<img width="1028" alt="Screen Shot 2021-08-09 at 13 53 57" src="https://user-images.githubusercontent.com/2406457/128744175-5ef66e93-8ac2-400d-a2c5-4f47b67abb8d.png">

Reply:
the `sdk.NewCoins` try to validate and causes this panic, I will create the tests in this way:

`sdk.Coins{sdk.Coin{Denom: "", Amount: sdk.NewInt(10)}}`

Reply:
You mean `sdk.NewCoins` makes this panic?

We can define the value manually then since `Coins` is an alias of `[]Coin`: `[]sdk.Coin{
{...}
}`

Reply:
yes. `sdk.NewCoins` cause panics
I created the tests directly with the object (`sdk.Coins{}`)  instead call the `sdk.NewCoins`

Reply:
also, should we be careful to use `sdk.NewCoins` to not crash the application if the user passes an invalid coin

Reply:
If it is called in `cli` it doesn't matter. We must avoid calling it in `keeper` and `types` but I doubt we call it there